File: src/main/java/io/eontimer/service/factory/timer/FixedFrameTimerFactory.java
package io.eontimer.service.factory.timer;

import java.util.Arrays;
import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import io.eontimer.service.CalibrationService;

@Service
public class FixedFrameTimerFactory {

	@Autowired
	private CalibrationService calibrationService;

	public List<Long> createStages(long preTimer, long targetFrame, long calibration) {
		return Arrays.asList(stage1(preTimer), stage2(targetFrame, calibration));
	}

	public long calibrate(long targetFrame, long frameHit) {
		return calibrationService.toMillis(targetFrame - frameHit);
	}

	private long stage1(long preTimer) {
		return preTimer;
	}

	private long stage2(long targetFrame, long calibration) {
		return calibrationService.toMillis(targetFrame) + calibration;
	}

}

File: src/service/factory/FixedFrameTimerFactory.js
import calibrationUtil from "../util/CalibrationUtil";

const createStages = (preTimer = 0, targetFrame = 0, calibration = 0, handheld) => {
  return [preTimer, calibrationUtil.toMillis(targetFrame, handheld) + calibration];
};

const calibrate = (targetFrame = 0, frameHit = 0, handheld) => {
  return calibrationUtil.toMillis(targetFrame - frameHit, handheld);
};

export default {
  createStages,
  calibrate,
};
